source-code/website/src/services/markdown/src/nodes/Heading.tsx

```diff
@@ -49,21 +49,60 @@ const Wrapper = (props: { children: JSXElement }) => {
 
 const CopyWrapper = (props: { children: JSXElement }) => {
 	const [isHovered, setIsHovered] = createSignal(false)
+	let element: HTMLDivElement | ((el: HTMLDivElement) => void) | undefined
+
 	return (
 		<div
 			class="relative cursor-pointer"
 			onMouseEnter={() => setIsHovered(true)}
 			onMouseLeave={() => setIsHovered(false)}
-			id={props.children?.toString().replace(" ", "-").toLowerCase()}
+			ref={(el) => {
+				if (typeof el === "function") {
+					element = el
+				} else if (el instanceof HTMLDivElement) {
+					element = el
+				}
+			}}
+			id={
+				props.children
+					?.toString()
+					.replace(" ", "-")
+					.replace("#", "")
+					.toLowerCase()
+					.includes("native code")
+					? typeof element === "object" && element instanceof HTMLDivElement
+						? element.innerText?.toString().replace(" ", "-").replace("#", "").toLowerCase().trim() // Trim the value to remove any leading/trailing whitespace
+						: ""
+					: props.children?.toString().replace(" ", "-").toLowerCase()
+			}
```

Comment:
can you make this inline code into its own function? maybe we need this logic elsewhere in the future.